metapool/sample_sheet.py

```diff
@@ -1370,52 +1407,32 @@ class AbsQuantSampleSheetv10(KLSampleSheet):
         'Chemistry': 'Default',
     }
 
-    data_columns = [SS_SAMPLE_ID_KEY, _SS_SAMPLE_NAME_KEY, 'Sample_Plate',
-                    'well_id_384', 'I7_Index_ID', 'index', 'I5_Index_ID',
-                    'index2', _SS_SAMPLE_PROJECT_KEY, 'mass_syndna_input_ng',
-                    'extracted_gdna_concentration_ng_ul',
-                    'vol_extracted_elution_ul', 'syndna_pool_number',
-                    'Well_description']
+    _data_columns = \
+        _BASE_DATA_COLUMNS + AbsQuantMixin._ABSQUANT_SPECIFIC_COLUMNS
 
-    _KL_ADDTL_DF_SECTIONS = {
+    _KL_ADDTL_DF_SECTIONS = MappingProxyType({
         _BIOINFORMATICS_KEY: _BIOINFORMATICS_COLS_W_REP_SUPPORT,
         _CONTACT_KEY: _CONTACT_COLS,
-    }
+    })
 
-    CARRIED_PREP_COLUMNS = [EXPT_DESIGN_DESC_KEY,
-                            'extracted_gdna_concentration_ng_ul',
-                            'i5_index_id', 'i7_index_id', 'index', 'index2',
-                            'library_construction_protocol',
-                            'mass_syndna_input_ng', SAMPLE_NAME_KEY,
-                            'sample_plate', 'sample_project',
-                            'syndna_pool_number', 'vol_extracted_elution_ul',
-                            'well_description', 'well_id_384']
+    _CARRIED_PREP_COLUMNS = \
+        _BASE_CARRIED_PREP_COLUMNS + AbsQuantMixin._ABSQUANT_SPECIFIC_COLUMNS
 
     def __init__(self, path=None):
         super().__init__(path=path)
-        self.remapper = {
-            'sample sheet Sample_ID': SS_SAMPLE_ID_KEY,
-            'Sample': _SS_SAMPLE_NAME_KEY,
-            PM_PROJECT_PLATE_KEY: 'Sample_Plate',
-            'Well': 'well_id_384',
-            'i7 name': 'I7_Index_ID',
-            'i7 sequence': 'index',
-            'i5 name': 'I5_Index_ID',
-            'i5 sequence': 'index2',
-            PM_PROJECT_NAME_KEY: _SS_SAMPLE_PROJECT_KEY,
-            'syndna_pool_number': 'syndna_pool_number',
-            'mass_syndna_input_ng': 'mass_syndna_input_ng',
-            'extracted_gdna_concentration_ng_ul':
-                'extracted_gdna_concentration_ng_ul',
-            'vol_extracted_elution_ul':
-                'vol_extracted_elution_ul'
-        }
+        self._remapper = AbsQuantMixin._ABSQUANT_REMAPPER
+
+
+class AbsQuantSampleSheetv11(AbsQuantMixin, KLSampleSheetWithSampleContext):
+    _HEADER = AbsQuantSampleSheetv10._HEADER.copy()
```

Comment:
Similar to my earlier comment regarding v101 and v102 sheets. In this case we do have some AbsQuantSampleSheetv10() sheets laying around, but since all v10 sheets would be valid v11 sheets as well, why not just add this functionality to v10?

Reply:
Well, no, v10 sample sheets are not also valid as v11 sample sheets.  You are probably remembering that there's a bunch of code in the `sample_sheet` module does not require the `SampleContext` section--but the code you might be thinking of is in `KLSampleSheet` and applies to all different versions of sample sheets.  That version-agnostic code does not require the presence of a `SampleContext` section so that it can continue to run on old sheets that don't include `SampleContext`. However, the `KLSampleSheetWithContext`-specific class definition enforces that any `KLSampleSheetWithContext`-based sheet *is* required to contain the `[SampleContext]` section header and its associated column headers line even if there are no actual records to go into that section.  

I settled on this requirement because it makes it explicit that a sheet containing a present-but-empty context section was generated by a process that knows about sample context but that no context happens to exists in this particular sheet, whereas if the context section were left out altogether in this situation, it would mean we couldn't tell by looking at a sample sheet whether it *had* no controls in it or whether we hadn't bothered to *look* for controls in it :)

Reply:
Very true!